### DISCUSSION TITLE
Development Mode, Held Messages... what happens after the hold period is over?

### DISCUSSION THREAD

**Discussion by JohnHammell** [postalserver/postal#3123] [Help with using Postal] [answered]
When a Postal server is in Development Mode and emails sent through that server's credentials are held, those emails have the following message displayed:

>This message has been held. By releasing the message, we will allow it to continue on its way to its destination. It will be held until September 11, 2024 20:31.

What happens when the "held until" time is reached?

What happens when the "Cancel Hold" button is clicked? I tested that and the message status became "Hold Cancelled" and it appears nothing else was done with the message.

I'm guessing that when the "held until" time is reached that the hold on the message is then cancelled, and nothing else. Basically the same as using the "Cancel Hold" button.

I just wanted to be certain that when the "held until" time is reached that wouldn't begin delivering emails from this development server set in dev mode or anything like that.

**Comment by willpower232** [comment, marked as answer]
> What happens when the "held until" time is reached?

The message can no longer be released

> What happens when the "Cancel Hold" button is clicked? I tested that and the message status became "Hold Cancelled" and it appears nothing else was done with the message.

Mostly it just gets the message off of the Held Messages page so you can deal with the other held messages

> I'm guessing that when the "held until" time is reached that the hold on the message is then cancelled, and nothing else. Basically the same as using the "Cancel Hold" button.

Correct

> I just wanted to be certain that when the "held until" time is reached that wouldn't begin delivering emails from this development server set in dev mode or anything like that.

You don't need to be worried, this isn't how Postal behaves. If your server in the UI is in dev mode or the credential is holding then it won't send messages. You can always manually release the message if you want to test the other half of things.

**Reply by JohnHammell** [reply]
Thanks for clarifying these things.

Also, I did find out something else while testing today. When we click "Cancel Hold" the "Release message" option does go away, but we then get a different option to "Redeliver message." If at that point we click on "Redeliver message" the email is literally delivered and not fake-delivered / held in dev mode. So the original delivery of the email on the dev mode server is Held and not delivered. But the "Redeliver message" option does deliver. I guess what that boils down to is that we can still (re)deliver a message after the hold is canceled and/or after the "held until" time period passes in case that is needed. Figured it might be good to also post that here as well in case it's useful for anyone else who comes across this support question.
